### PR TITLE
[ADF-4824] Add AuthGuard

### PR DESCRIPTION
**Pull request by mmuller88**
Issue Number: ADF-4824

### CODE REVIEW

**Comment by DenysVuika** [comment]
@mmuller88 I have updated the PR header to match the convention we use

File: src/app/app.routes.ts [hunk 1]
```diff
@@ -73,6 +73,7 @@ export const APP_ROUTES: Routes = [
   {
     path: '',
     component: AppLayoutComponent,
+    canActivate: [AuthGuard],
```
Comment (pionnegru) [inline src/app/app.routes.ts]: can we keep both guards in the same place?

**Comment by suzanadirla** [comment]
hi.. I see that the AuthGuard does a call to ` this.redirectToUrl('ALL', redirectUrl);` at some point, and ACA has 'ECM' set as the provider inside app.copnfig.json file. I wonder if that causes issues.

**Comment by suzanadirla** [comment]
After more investigating the code: It seems if 'ALL' covers either 'ECM' or 'BPM' or both - so all should be ok.

File: src/app/app.routes.ts (hunk 1 above)
Comment (mauriziovitale) [reply src/app/app.routes.ts]: I don't think is needed, the AuthGuard should be smart enough to check if a user is logged in SSO/BPM/ECM

**Pull request merged**